src/main/scala/group/holmes/analytics/actors/Scheduler.scala

```diff
@@ -1,19 +1,93 @@
 package group.holmes.analytics.actors
 
 import java.io.File
+import java.util.UUID
+
+import scala.collection.mutable.HashMap
+import scala.concurrent.duration._
+import scala.concurrent.Await
 
 import akka.actor.{ Actor, ActorLogging, ActorRef, Props }
+import akka.pattern.ask
+import akka.util.Timeout
 import com.typesafe.config.ConfigFactory
 
-
 object Scheduler {
 	def props(analyticEngineManager: ActorRef, analyticServiceManager: ActorRef): Props = Props(new Scheduler(analyticEngineManager, analyticServiceManager))
+
+	// Messages
+	// Message used to create a new Job Actor
+	final case class New(name: String, engine: String, service: String, parameters: HashMap[String, String])
+
+	// Message used to get the status of a Job
+	final case class Status(id: UUID)
+
+	// Force a refresh of all job statuses
+	final case class Refresh()
+
+	// Get a list of all jobs
+	final case class List()
+
+	//Exposed objs
+	final case class ScheduledJob(ref: ActorRef, id: UUID, name: String, status: String)
 }
 
 class Scheduler(analyticEngineManager: ActorRef, analyticServiceManager: ActorRef) extends Actor with ActorLogging {
+	private val jobs = HashMap.empty[UUID, Scheduler.ScheduledJob]
+	implicit val timeout: Timeout = 10.seconds //TODO: Discuss sensible value / error mitigation strategy
```

Comment:
Might be a good idea, so far I'm debating between a hard value and super-visioned restart vs. long timeout.